aten/src/ATen/templates/TensorBody.h

```diff
@@ -1157,4 +1165,103 @@ inline c10::MaybeOwned<Tensor> Tensor::expect_contiguous(MemoryFormat memory_for
     return c10::MaybeOwned<Tensor>::owned(__dispatch_contiguous(memory_format));
   }
 }
+
```

Comment:
Note that we already have a class called `ViewInfo` that is stored on the `AutogradMeta` (https://github.com/pytorch/pytorch/blob/87a661c79f486f2fcc11f125a86a2b5f0ec53d83/torch/csrc/autograd/variable.h#L289) this might lead to some confusion in naming.

Reply:
I namespaced this into `at::functionalization::ViewMeta` in the new PR